net/golioth/golioth.c

```diff
@@ -299,7 +299,11 @@ static int golioth_sendmsg(struct golioth_client *client,
 
 	free(data);
 
-	return ret;
+	if (ret < len) {
+		return -EIO;
```

Comment:
Could return `ret` here instead, since that would pick up the `errno` (in case the send call set an error other than EIO).

Alternatively, could split it into two cases:

```c
if (ret < 0) {
    return ret;
} else if (ret < len) {
    return -EIO;
}
```

Reply:
Implemented the "alternative" version. EIO will be returned in case not full DGRAM was sent at once (in theory Zephyr should not allow that, but I have doubts that is the case).